platform/source/mbed_alloc_wrappers.cpp

```diff
@@ -347,6 +347,16 @@ extern "C" void *SUB_REALLOC(void *ptr, size_t size)
         memcpy(new_ptr, (void *)ptr, copy_size);
         free(ptr);
     }
+
+    {
+        volatile uint8_t dummy = 0;
+        if (dummy != 0) { // always false
+            // this code will never be executed
+            // it's just to tell the compiler/linker to preserve SUB_REALLOC symbol
+            // when LTO enabled
+            SUPER_REALLOC(NULL, 0);
```

Comment:
Maybe I'm being dense, but why aren't these calls using `SUPER_REALLOC` and `SUPER_CALLOC` anyway? Why are they messing around re-implementing them?

Reply:
Okay, I see the point - if you did use them, you'd have to reimplement the stuff in your malloc wrapper. Probably a bit bigger overall.

Reply:
I'm not sure how that fix actually works, mind - so you've inserted a (sort-of) call to the $$Super, but how does that persuade the LTO to retain the $$Sub?

Or maybe that activates a magic heuristic in the compiler - if a function calls any $$Super, assume it's referenced? Hmm. But then why not treat any $$Sub as referenced?

Can you explain how you got to this?

Reply:
@jamesbeyond found that difference between `SUB_CALLOC/REALOC` and `SUB_MALLOC/FREE`

And then by the analogy I have added dummy `SUPER_CALLOC/REALOC` calls inside `SUB_CALLOC/REALOC`